src/test/java/pages/MainPage.java

```diff
@@ -68,7 +103,96 @@ public String getLoginViaNickNameMessage() {
      * Display message "Cart" while entering the shopping cart
      * @return return message of shoppingCartMessage element
      */
-    public String getShoppingCartMessage() {
-        return shoppingCartMessage.getText().toLowerCase().trim();
+    public String getLoginWarningMessage() {
+        WaitUtils.waitForVisibility(incorrectCredentialsWarning);
+        return incorrectCredentialsWarning.getText().toLowerCase().trim();
+    }
+
+    /***
+     * While entering no password, a warning message appears
+     * @return warning message
+     */
+    public String getPasswordInputWarning() {
+        WaitUtils.waitForVisibility(emptyPasswordWarning);
+        return emptyPasswordWarning.getText().toLowerCase().trim();
+    }
+
+    /***
+     * While entering a phrase into the input field, a hint appears
+     * @return hint message
+     */
+    public String getSearchInputHint() {
+        driver.switchTo().frame(driver.findElement(By.className("modal-iframe")));
+        WaitUtils.waitForVisibility(searchInputHint);
+        return searchInputHint.getText().toLowerCase().trim();
+    }
+
+    /***
+     * Insert the login into the login input
+     * @param login login
+     */
+    public MainPage insertLogin(String login) {
+        loginInput.sendKeys(login);
+        return this;
+    }
+
+    /***
+     * Insert the password to into the password input
+     * @param psw password
+     */
+    public MainPage insertPassword(String psw) {
+        passwordInput.sendKeys(psw);
+        return this;
+    }
+
+    /***
+     * Click the entrance button
+     */
+    public MainPage clickEntranceButton() {
+        loginButton.click();
+        return this;
+    }
+
+    /***
+     * Insert a text into the search field and look for items
+     * @param text any query
+     */
+    public MainPage insertTextIntoSearch(String text) {
+        mainSearchField.sendKeys(text);
+        mainSearchField.sendKeys(Keys.ENTER);
+        return this;
+    }
+
+    /***
+     * Click on the catalogue button
+     */
+    public CataloguePage clickOnCatalogue() {
+        catalogue.click();
+        return new CataloguePage();
+    }
+
+    /***
+     * When signed in, an avatar appears on the main page
+     */
+    public boolean getAvatar() {
+        WaitUtils.waitForVisibility(avatar);
+        avatar.isDisplayed();
+        return true;
```

Comment:
why do you always return true in any condition? MAybe it makes sense to update it to 'return avatar.isDisplayed()' ?

Reply:
changed